uitools/cpp/Esri/ArcGISRuntime/Toolkit/UtilityNetworkTraceStartingPointsModel.h

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#ifndef UTILITYNETWORKTRACESTARTINGPOINTSMODEL_H
+#define UTILITYNETWORKTRACESTARTINGPOINTSMODEL_H
+
+#include <QAbstractListModel>
+
+// ArcGISRuntime headers
+#include <Point.h>
+
+namespace Esri::ArcGISRuntime {
+
+class UtilityElement;
+
+namespace Toolkit {
+class UtilityNetworkTraceStartingPoint;
+
+class UtilityNetworkTraceStartingPointsModel : public QAbstractListModel
```

Comment:
Do you think that the `GenericListModel` is not generic enough since you had to re-implement something similar by inheriting `QAbstractListModel`?

Reply:
I am not entirely sure how to use the `GenericListModel`, especially with regards to setting up the roles and managing their manipulation (`setData()`, `data()`)